### ISSUE TITLE
org.dbunit.dataset.NoPrimaryKeyException when using @ElementCollection in @Entity

### ISSUE THREAD

**Issue by pmajkutewicz** [database-rider/database-rider#249]
Having (more or less) class like:

```
@Entity
public class Person implements Serializable {

	@Column
	private Integer id;

	@Column
	private String name;

	@ElementCollection(targetClass = String.class, fetch = FetchType.EAGER)
	@CollectionTable(name="nick_names", joinColumns=@JoinColumn(name="p_id"))
	@Column(name="nick")
	private List<String> nickNames;
}
```

And dataset like:

```
<?xml version="1.0" encoding="UTF-8"?>
<dataset>
	<person id="1" name="John"/>
	<nick_names p_id="1" nick="Johny"/>
</dataset>
```

After running test that uses this dataset I'm getting:

```
java.lang.RuntimeException: Could not create dataset for test 'x'.

	at com.github.database.rider.core.RiderRunner.runBeforeTest(RiderRunner.java:47)
	at com.github.database.rider.spring.DBRiderTestExecutionListener.beforeTestMethod(DBRiderTestExecutionListener.java:25)
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:289)
	at org.springframework.test.context.junit.jupiter.SpringExtension.beforeEach(SpringExtension.java:109)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachCallbacks$1(TestMethodTestDescriptor.java:159)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$5(TestMethodTestDescriptor.java:195)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeMethodsOrCallbacksUntilExceptionOccurs(TestMethodTestDescriptor.java:195)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeEachCallbacks(TestMethodTestDescriptor.java:158)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:125)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: com.github.database.rider.core.exception.DataBaseSeedingException: Could not initialize dataset: datasets/common.xml, datasets/x/y.xml
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.createDataSet(DataSetExecutorImpl.java:173)
	at com.github.database.rider.core.RiderRunner.runBeforeTest(RiderRunner.java:45)
	... 52 more
Caused by: org.dbunit.dataset.NoPrimaryKeyException: NICK_NAMES
	at org.dbunit.operation.UpdateOperation.getOperationData(UpdateOperation.java:72)
	at org.dbunit.operation.RefreshOperation$UpdateRowOperation.<init>(RefreshOperation.java:275)
	at org.dbunit.operation.RefreshOperation.createUpdateOperation(RefreshOperation.java:151)
	at org.dbunit.operation.RefreshOperation.execute(RefreshOperation.java:103)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.createDataSet(DataSetExecutorImpl.java:164)
	... 53 more
```

Founded article about it: https://dzone.com/articles/dbunit-unable-import-data but I don't know (can't find in docs) how to set up `IColumnFilter` with database rider.

**Comment by rmpestano** [comment]
Hi @pmajkutewicz, 

> but I don't know (can't find in docs) how to set up IColumnFilter with database rider.

I don't think we have a way to add a Column filter dynamically, I think we would need to have a property in `@DataSet` to enable the loading of datasets without a primary key.

I'll try to work on this for the next release and let you know when it's ready so you can try it.

Thank you for the feedback!

**Comment by rmpestano** [comment]
Hi @pmajkutewicz, 

I've added a new property in `@DBUnit` called `disablePKCheckFor` where you can specify the tables you don't want the PK check:

```
    @Test
    @DataSet(value = "yml/nopk.yml", cleanBefore = true, strategy = SeedStrategy.REFRESH)
    @DBUnit(disablePKCheckFor = "NOPK")
    public void shouldSeedTableWithoutPK() {
    }
``` 

You can also configure it globally via `dbunit.yml` file.

You can try it now by using `1.20.1-SNAPSHOT`, to do so you need to enable [snapshots repo ](https://github.com/database-rider/database-rider#snapshots)on your build system.

Let me know if it works for you so we can release it in `v1.21.0`.

**Comment by rmpestano** [comment]
It's available in just released version `1.21.0`.

**Comment by pmajkutewicz** [comment]
@rmpestano 
I'm not so sure if this is working...
I've added ` @DBUnit(disablePKCheckFor = "FEATURES")` (generated by @ElementCollection(targetClass = String.class, fetch = FetchType.EAGER)) and now I'm getting 

```
Caused by: org.dbunit.dataset.NoPrimaryKeyException: COMPANY
	at org.dbunit.operation.UpdateOperation.getOperationData(UpdateOperation.java:72)
	at org.dbunit.operation.RefreshOperation$UpdateRowOperation.<init>(RefreshOperation.java:275)
	at org.dbunit.operation.RefreshOperation.createUpdateOperation(RefreshOperation.java:151)
	at org.dbunit.operation.RefreshOperation.execute(RefreshOperation.java:103)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.createDataSet(DataSetExecutorImpl.java:163)
```	
For every other table (with PKs) - so basically guessing that I have to put all my tables in `disablePKCheckFor`.
